src/main/java/com/puppycrawl/tools/checkstyle/GlobalStatefulModule.java

```diff
@@ -0,0 +1,33 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle;
+
+/**
+ * This interface means that the module contains global context,
+ * which will be updated accross all the application lifetime.
+ * This annotation could be used, if some information should be accumulated across the application
+ * lifetime, and processed only once at the end of application lifetime.
```

Comment:
> processed only once at the end of application lifetime.

Besides TranslationCheck, all others don't wait to the end of the application right?
So this statement is really true. The check may produce violations before the end, if it so chooses.